test/integration/daemonset/daemonset_test.go

```diff
@@ -875,7 +879,20 @@ func TestLaunchWithHashCollision(t *testing.T) {
 
 	// Make an update of the DaemonSet which we know will create a hash collision when
 	// the next ControllerRevision is created.
-	_, err = dsClient.Update(ds)
+	dsBytes, err := json.Marshal(ds)
+	if err != nil {
+		t.Fatal(err)
+	}
+	oldDsBytes, err := json.Marshal(oldDs)
+	if err != nil {
+		t.Fatal(err)
+	}
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldDsBytes, dsBytes, ds)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = dsClient.Patch(ds.Name, types.StrategicMergePatchType, patchBytes)
```

Comment:
Thanks for getting back here, at least retrospectively.

I guess for this particular use case that would work, but I don't like the pattern in general though. If you have the object from previous step, there is no reason it should do the `GET` call for you. Also the whole `updateRS` func is there to deal with `RetryOnConflict` and we don't need it when using `PATCH`. We should be good just with:

```golang
	oldRS, rs := rs, rs.DeepCopy()
	rs.Spec.Replicas = 5
	patchBytes, err := getPatchBytes(oldRS, rs)
	if err != nil {
		t.Fatalf("Failed to update rs %s: %v", rsName, err)
	}
	rs, err = rsClient.Patch(rsName, types.MergePatchType, patchBytes)
	if err != nil {
		t.Fatalf("Failed to update rs %s: %v", rsName, err)
	}
```

Writing small patches by hand would work as well, although you'd loose the semantics check. On the other hand you don't need to even have the object (`GET`) in the first place.

Maybe server side apply will give use a nicer interface.

In any case I don't want to bikeshed on tests but given this was ready before the other PR, and linked there, calling the semantically correct API method might be something to consider before merging.

Reply:
I wanted to de-flake the test first and discuss the details later. 

I like using patch to avoid the need to retry update. However, in tests, whenever the author needs to update something, they need to:
```go
    oldDs, ds := ds, ds.DeepCopy()
    // Do something here to update `ds`
	dsBytes, err := json.Marshal(ds)
	if err != nil {
		t.Fatal(err)
	}
	oldDsBytes, err := json.Marshal(oldDs)
	if err != nil {
		t.Fatal(err)
	}
	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldDsBytes, dsBytes, ds)
	if err != nil {
		t.Fatal(err)
	}

	_, err = dsClient.Patch(ds.Name, types.StrategicMergePatchType, patchBytes)
	if err != nil {
		t.Fatalf("Failed to update DaemonSet: %v", err)
	}
```

and I'd like to simplify this to:
```go
ds, err = updateDS(t, client, ds.Name, func(updateDS *apps.DaemonSet) {
		// Do something here to update ds
})
```